Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/NetworkHelper.kt

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package org.dolphinemu.dolphinemu.utils
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.LinkAddress
+import android.os.Build
+import androidx.annotation.Keep
+import androidx.annotation.RequiresApi
+import org.dolphinemu.dolphinemu.DolphinApplication
+import java.net.Inet4Address
+import java.net.InetAddress
+import java.net.UnknownHostException
+
+@Keep
+object NetworkHelper {
+    private fun getConnectivityManager(): ConnectivityManager? {
+        val context = DolphinApplication.getAppContext()
+        return (context.getSystemService(Context.CONNECTIVITY_SERVICE) as? ConnectivityManager)
+            ?: run {
+                Log.warning("Cannot get network link as ConnectivityManager is null.")
+                null
+            }
```

Comment:
Personally I think this construct is less readable than the assignment to variable and if statement that we had in Java.